fe-public/src/components/Calendar.astro

```diff
@@ -0,0 +1,37 @@
+---
+import ical from 'node-ical';
+let URL = import.meta.env.ICAL || "https://ieconnects.ie.edu/ical/ical_club_300003041.ics"
+const webEvents = await ical.async.fromURL(URL);
+let events = Object.values(webEvents).map((event) => {
+  return {
+    title: event.summary?.val,
+    start: new Date(event.start),
+    url: event.url,
+    description: event.description
+  };
+})
+events = events.sort((a, b) => {
+  return a.start - b.start;
+}).filter((event) => {
+  return event.start > new Date();
+});
+---
+<h1>Upcoming Events</h1>
+<table>
+<thead>
+    <tr>
+        <th>Title</th>
+        <th>Start</th>
+        <th>URL</th>
+    </tr>
+</thead>
+<tbody>
+{events.map((event) => (
+  <tr>
+    <td>{event.title}</td>
+    <td>{event.start}</td>
```

Comment:
It would be great to remove the date, not showing it just as a field but have it work sorta as a key. Where we have a linear progression top-to-bottom with the date being the highlighted key - pairing with the event name, which the user can click to goto the link.

Reply:
Like at the airport it is always by departure time, then the flight, then the gate ahahha